examples/t_factory.py

```diff
@@ -0,0 +1,94 @@
+import numpy as np
+
+from guppylang.decorator import guppy
+from guppylang.module import GuppyModule
+from guppylang.prelude.builtins import linst, py
+from guppylang.prelude.quantum import (
+    cz,
+    discard,
+    h,
+    measure,
+    quantum,
+    qubit,
+    rx,
+    rz,
+)
+
+module = GuppyModule("tfactory")
+module.load(quantum)
+
+phi = np.arccos(1 / 3)
+pi = np.pi
+
+
+@guppy(module)
+def ry(q: qubit, theta: float) -> qubit:
+    q = rx(q, py(pi / 2))
+    q = rz(q, theta + py(pi))
+    q = rx(q, py(pi / 2))
+    return rz(q, py(pi))
+
+
+# Preparation of approximate T state, from 2310.12106
```

Comment:
If an arxiv link, suggest you put in the full address (at least `arxiv`, preferably hostname and path, possibly protocol `https://`)